docs/release.md

```diff
@@ -0,0 +1,40 @@
+## Overview: `release.yaml`
```

Comment:
Should we reference this file from `README.md`?

Reply:
I don't think so. I think the readme should be more focused on usage, and the release pipeline is only relevant to core contributors.